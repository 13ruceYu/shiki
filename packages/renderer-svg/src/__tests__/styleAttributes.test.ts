import { getSVGRenderer } from '../index'

test('SVG renderer should generate SVG with correct style attributes', async () => {
  const r = getSVGRenderer({
    fontFamily: 'IBM Plex Mono'
  })

  const out = (await r).renderToSVG([[{ content: 'foo', color: '#aabbccff' }]])
  // 0x40 = 64, 0xff = 256. 64/256 = 0.25
  expect(out).toContain('opacity="1"')
})

test('SVG can have 0 corner radius', async () => {
  const r = getSVGRenderer({
    fontFamily: 'IBM Plex Mono',
    bgCornerRadius: 0
  })

  const out = (await r).renderToSVG([[{ content: 'foo', color: '#aabbccff' }]])
  expect(out).toContain('rx="0"')
})
