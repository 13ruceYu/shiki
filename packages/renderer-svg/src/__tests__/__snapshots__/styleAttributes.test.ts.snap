// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG renderer should ignore minimal background width when it's own boundaries are above that 1`] = `
"<svg viewBox=\\"0 0 156.4921875 101.28124999999999\\" width=\\"156.4921875\\" height=\\"101.28124999999999\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<rect id=\\"bg\\" fill=\\"#fff\\" width=\\"156.4921875\\" height=\\"101.28124999999999\\" rx=\\"4\\"></rect><g id=\\"tokens\\" transform=\\"translate(56.90625, 40.512499999999996)\\"><text font-family=\\"IBM Plex Mono\\" font-size=\\"16\\" y=\\"20.256249999999998\\">
<tspan x=\\"0\\" fill=\\"#aabbcc\\" opacity=\\"1\\">foo</tspan>
</text></g>
</svg>
"
`;
